server/app/constants/common.ts

```diff
@@ -1,3 +1,5 @@
 const NON_AUTH_PATHS: string[] = ['/user/login', '/user/register'];
 
-export { NON_AUTH_PATHS };
+const PATHS_FORBIDDEN_TO_GENERAL_ROLE: string[] = ['/user'];
```

Comment:
Non blocking suggestion, I suspect the amount of endpoints that will be forbidden for general role will be small, so you can create another middleware function called checkAdminRole and use it only on endpoints that need it. Example given below

Reply:
Ya that could work for now since we only have two roles but maybe later a buyer can do something that a seller can't do. We're prob gonna need to update the middleware so ill separate it from auth.